### PR TITLE
BaSP-MR-09-23: Update Class

### PR DESCRIPTION
**Pull request by josedaniele**
The put method to update class data has been implemented, and the schema has been updated to change the trainer data from a name to an ID

### CODE REVIEW

**Review by NicoCagnina — APPROVED** [review]

File: src/validations/class.js [hunk 1]
```diff
@@ -12,4 +12,37 @@ const validateId = (req, res, next) => {
...
-module.exports = { validateId };
+const validateUpdate = (req, res, next) => {
+  const classValidation = Joi.object({
+    hour: Joi.string()
+      .pattern(/^([01]?[0-9]|2[0-3]):([0-5][0-9])$/)
+      .messages({
+        'string.pattern.base':
+        'The field must be a valid hour(for example 09:30 )',
+      }),
+    day: Joi.string()
```
Comment (JuanMornik) [inline src/validations/class.js:23]: Day must be an enum with the days of the week.
You also have to change the message and the Class schedule

File: src/models/Class.js [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
+const classSchema = new Schema({
+  hour: {
+    type: String,
+    required: true,
+    match: /^([01]?[0-9]|2[0-3]):([0-5][0-9])$/,
+  },
+  day: {
+    type: String,
+    required: true,
+    match: /^(0?[1-9]|[12][0-9]|3[01])\/(0?[1-9]|1[0-2])\/\d{4}$/,
```
Comment (JuanMornik) [inline src/models/Class.js:14]: Delete this line and update classSchema according to changes indicated on validations file

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/validations/class.js [hunk 3]
```diff
@@ -12,4 +12,33 @@ const validateId = (req, res, next) => {
...
+      .valid('Monday', 'Tuesday', 'Wendesday', 'Thursaday', 'Friday', 'Saturday', 'Sunday'),
```
Comment (JuanMornik) [inline src/validations/class.js:24]: Replace Wendesday with Wednesday

**Review by JuanMornik — CHANGES_REQUESTED** [review]

File: src/validations/class.js (hunk 3 above)
Comment (JuanMornik) [reply src/validations/class.js:24]: And add a message too

**Review by JuanMornik — APPROVED** [review]

**Pull request merged**